### ISSUE TITLE
Support notifications api

### ISSUE THREAD

**Issue by GUIpsp** [jcabi/jcabi-github#913]

**Comment by nathansgreen** [comment]
@dmarkov I'm not seeing my login as a label on this issue. I totally missed this. Please assign to someone else.

**Comment by dmarkov** [comment]
> @dmarkov I'm not seeing my login as a label on this issue. I totally missed this. Please assign to someone else.

@nathansgreen someone else will help in this task, no problem at all

**Comment by dmarkov** [comment]
@nathansgreen you're working with this issue for 13 days already... -30 added to your rating, at the moment it is: [+70](http://www.netbout.com/b/35107?open=rating)

**Comment by dmarkov** [comment]
@dpisarenko it's your task, go ahead

**Comment by dpisarenko** [comment]
If I understand correctly, we have following API calls (source: https://developer.github.com/v3/activity/notifications/ ):

  1. List your notifications - `com.jcabi.github.User#notifications`
  1. List your notifications in a repository - `com.jcabi.github.Notifications#iterate`
  1. Mark as read - `com.jcabi.github.User#markAsRead`
  1. Mark notifications as read in a repository - `com.jcabi.github.Notifications#markAsRead` 
  1. View a single thread - `com.jcabi.github.Notifications#thread`
  1. Mark a thread as read - `com.jcabi.github.GitHubThread#markAsRead`
  1. Get a Thread Subscription - `com.jcabi.github.GitHubThread#getSubscription`
  1. Set a Thread Subscription - `com.jcabi.github.GitHubThread#setSubscription`
  1. Delete a Thread Subscription - `com.jcabi.github.GitHubThread#deleteSubscription`

Next to each call you can find the location in the code, where I would put them.

@yegor256 @GUIpsp Please look at this list, the PR https://github.com/jcabi/jcabi-github/pull/985 and tell me, whether you agree with that design. Thanks!

**Comment by GUIpsp** [comment]
I think that 7-8 should use an enum rather than an interface.

Other than that looks OK.

**Comment by dpisarenko** [comment]
@GUIpsp Interface `ThreadSubscription` is supposed to contain data in response of *Get a Thread Subscription* ( https://developer.github.com/v3/activity/notifications/#get-a-thread-subscription ) and *Set a Thread Subscription* ( https://developer.github.com/v3/activity/notifications/#set-a-thread-subscription ), namely:

```
{
  "subscribed": true,
  "ignored": false,
  "reason": null,
  "created_at": "2012-10-06T21:34:12Z",
  "url": "https://api.github.com/notifications/threads/1/subscription",
  "thread_url": "https://api.github.com/notifications/threads/1"
}
```

I don't see a way to put these data into an enum.

**Comment by GUIpsp** [comment]
Ah, I see. I assumed it was only for the subscribed/ignored part. should I close this issue?

**Comment by dpisarenko** [comment]
@GUIpsp No, we arent't done with the code review yet.

**Comment by dpisarenko** [comment]
@GUIpsp The PR has been merged. Unless there is reason no to do so, please close the issue.

**Comment by dmarkov** [comment]
@dpisarenko this task is in your hands for 15 days already; -30 added to your rating, current score is: [+656](http://www.netbout.com/b/35089?open=rating)

**Comment by dpisarenko** [comment]
@GUIpsp The PR has been merged. Unless there is reason no to do so, please close the issue.

**Comment by 0pdd** [comment]
@GUIpsp 9 puzzles [#unknown](), [#unknown](), [#unknown](), [#unknown](), [#unknown](), [#unknown](), [#unknown](), [#unknown](), [#unknown]() are still not solved; solved: [#920](https://github.com/jcabi/jcabi-github/issues/920).

**Comment by 0pdd** [comment]
@GUIpsp 8 puzzles [#unknown](), [#unknown](), [#unknown](), [#unknown](), [#unknown](), [#unknown](), [#unknown](), [#unknown]() are still not solved; solved: [#920](https://github.com/jcabi/jcabi-github/issues/920).

**Comment by 0pdd** [comment]
@GUIpsp 3 puzzles [#unknown](), [#unknown](), [#unknown]() are still not solved.
